CoffeeShop/src/main/java/client/CustomerController.java

```
@@ -3,165 +3,154 @@
-                return;
+            boolean orderAdded = model.submitOrder();
+            if (orderAdded) {
+                Demo.demoWriteOrders();  // Existing demo functionality
+                JOptionPane.showMessageDialog(view, "Order has been submitted");
+                updateView();
+            } else {
+                JOptionPane.showMessageDialog(view,
+                        "Order could not be placed - Please Try Again Later");
             }
-            Demo.demoWriteOrders();
-            JOptionPane.showMessageDialog(view, "Order has been submitted");
         } catch (InvalidOrderException | DuplicateOrderException e) {
             JOptionPane.showMessageDialog(view, e.getMessage());
         }
-        try {
-            currentOrder = new Order();
-            updateView();
-        } catch (InvalidOrderException e) {
-            throw new RuntimeException(e);
-        }
     }
 
     /**
-     * Cancels the current order
+     * Handles order cancellation
      */
-    private void cancelOrder() {
+    private void handleCancelOrder() {
+        model.cancelOrder();
         JOptionPane.showMessageDialog(view, "Order Cancelled");
-        try {
-            currentOrder = new Order();
-            updateView();
-        } catch (InvalidOrderException ex) {
-            throw new RuntimeException(ex);
-        }
+        updateView();
     }
 
     /**
-     * Adds an item to the current order
+     * Handles item addition
      */
-    private void addItem() {
-        String itemID = view.getItemIDField().getText();
-        try {
-            currentOrder.addItem(itemID.toUpperCase());
-        } catch (InvalidItemIDException e) {
-            JOptionPane.showMessageDialog(view, itemID.toUpperCase() + " is not a valid item ID");
+    private void handleAddItem() {
+        String itemID = view.getItemIDField().getText().trim();
+        if (!itemID.isEmpty()) {
```

can we invert this please? So add a guard clause (check if it is empty and then return if it is) so we fail fast